src/Avalonia.Visuals/Media/HsvColor.cs

```diff
@@ -0,0 +1,515 @@
+            return new HsvColor(h, s, v, a);
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="HsvColor"/> from individual color channel values.
+        /// </summary>
+        /// <remarks>
+        /// This exists for symmetry with the <see cref="Color"/> struct; however, the
+        /// appropriate constructor should commonly be used instead.
+        /// </remarks>
+        /// <param name="h">The Hue channel value in the range from 0..360.</param>
+        /// <param name="s">The Saturation channel value in the range from 0..1.</param>
+        /// <param name="v">The Value channel value in the range from 0..1.</param>
+        /// <returns>A new <see cref="HsvColor"/> built from the individual color channel values.</returns>
+        public static HsvColor FromHsv(double h, double s, double v)
+        {
+            return new HsvColor(h, s, v, 1.0);
+        }
+
+        /// <summary>
+        /// Converts the given HSV color to it's RGB color equivalent.
+        /// </summary>
+        /// <param name="hsvColor">The color in the HSV color model.</param>
+        /// <returns>A new RGB <see cref="Color"/> equivalent to the given HSVA values.</returns>
+        public static Color ToRgb(HsvColor hsvColor)
+        {
+            return HsvColor.ToRgb(hsvColor.H, hsvColor.S, hsvColor.V, hsvColor.A);
+        }
+
+        /// <summary>
+        /// Converts the given HSVA color channel values to it's RGB color equivalent.
+        /// </summary>
+        /// <param name="hue">The hue channel value in the HSV color model in the range from 0..360.</param>
+        /// <param name="saturation">The saturation channel value in the HSV color model in the range from 0..1.</param>
+        /// <param name="value">The value channel value in the HSV color model in the range from 0..1.</param>
+        /// <param name="alpha">The alpha channel value in the range from 0..1.</param>
+        /// <returns>A new RGB <see cref="Color"/> equivalent to the given HSVA values.</returns>
+        public static Color ToRgb(
+            double hue,
+            double saturation,
+            double value,
+            double alpha = 1.0)
+        {
+            // Note: Conversion code is originally based on the C++ in WinUI (licensed MIT)
+            // https://github.com/microsoft/microsoft-ui-xaml/blob/main/dev/Common/ColorConversion.cpp
+            // This was used because it is the best documented and likely most optimized for performance
+            // Alpha channel support was added
+
+            // We want the hue to be between 0 and 359,
+            // so we first ensure that that's the case.
+            while (hue >= 360.0)
+            {
+                hue -= 360.0;
+            }
+
+            while (hue < 0.0)
+            {
+                hue += 360.0;
+            }
+
+            // We similarly clamp saturation, value and alpha between 0 and 1.
+            saturation = saturation < 0.0 ? 0.0 : saturation;
```

Comment:
Could use `MathUtilities.Clamp` three times to remove some boilerplate.

Reply:
Might make sense to decorate MathUtilities.Clamp with `[MethodImpl(MethodImplOptions.AggressiveInlining)]` as System.Math does

Reply:
MathUtilities.Clamp is still called 4 times in the constructor at the end of this method. I'm not sure I can remove this safely as HsvColor needs to have validation to ensure parameter ranges. However, there is more overhead in doing this that isn't an issue in other types like Color where the parameter type can never be out of range.

Reply:
As a further optimization I added an internal-only constructor that can bypass channel parameter value clamping. This means MathUtilties.Clamp is never called in this method. Again, performance concerns in future color spectrum calculations.